api/src/main/java/com/messagebird/MessageBirdClient.java

```diff
@@ -1587,4 +1569,47 @@ private void verifyOffsetAndLimit(Integer offset, Integer limit) {
             throw new IllegalArgumentException("Limit must be > 0");
         }
     }
-}
\ No newline at end of file
+
+    public PhoneNumbersResponse listNumbersForPurchase(String countryCode) throws IllegalArgumentException, GeneralException, UnauthorizedException, NotFoundException {
+        final String url = String.format("%s/v1/available-phone-numbers", NUMBERS_CALLS_BASE_URL);
+        return messageBirdService.requestByID(url, countryCode, PhoneNumbersResponse.class);
+    }
+
+    public PhoneNumbersResponse listNumbersForPurchase(String countryCode, PhoneNumbersLookup params) throws IllegalArgumentException, GeneralException, UnauthorizedException, NotFoundException {
+        final String url = String.format("%s/v1/available-phone-numbers", NUMBERS_CALLS_BASE_URL);
+        return messageBirdService.requestByID(url, countryCode, params.toHashMap(), PhoneNumbersResponse.class);
+    }
+
+    public PurchasedNumberCreatedResponse purchaseNumber(String number, String countryCode, int billingIntervalMonths) throws UnauthorizedException, GeneralException {
+        final String url = String.format("%s/v1/phone-numbers", NUMBERS_CALLS_BASE_URL);
+
+        final Map<String, Object> payload = new LinkedHashMap<String, Object>();
+        payload.put("number", number);
+        payload.put("countryCode", countryCode);
+        payload.put("billingIntervalMonths", billingIntervalMonths);
```

Comment:
can you check billingIntervalMonths and return a meaningful exception.  According to doc, Valid values: 1, 3, 6, 9, 12